tests/a_scripts/data_prep/test_split_eval_set.py

```diff
@@ -0,0 +1,217 @@
+import os
+import json
+import pytest
+import hashlib
+from unittest.mock import patch
+
+from llmfoundry.command_utils import split_eval_set_from_args, split_examples
+from llmfoundry.command_utils.data_prep.split_eval_set import DELTA_JSONL_REGEX, get_dataset_format, HF_REGEX, REMOTE_OBJECT_STORE_FILE_REGEX
+
+# Default values
+OUTPUT_DIR = "tmp-split"
+TMPT_DIR = "tmp-t"
+DATA_PATH_SPLIT = "train"
+EVAL_SPLIT_RATIO = 0.1
+DEFAULT_FILE = TMPT_DIR + "/train-00000-of-00001.jsonl"
+
+
+def test_delta_jsonl_regex():
+    """Test the regex pattern matches tmp-t exactly"""
+    assert DELTA_JSONL_REGEX.match("tmp-t")
+    assert not DELTA_JSONL_REGEX.match("/tmp-t")
+    assert not DELTA_JSONL_REGEX.match("tmp-t-00000-of-00001.jsonl")
+    assert not DELTA_JSONL_REGEX.match("tmp-t-something")
+    assert not DELTA_JSONL_REGEX.match("tmp-t/")
+    assert not DELTA_JSONL_REGEX.match("tmp-t\\")
+
+def test_remote_object_store_file_regex():
+    """Test the regex pattern for remote object store file paths"""
+    assert REMOTE_OBJECT_STORE_FILE_REGEX.match("s3://bucket-name/path/to/file")
+    assert REMOTE_OBJECT_STORE_FILE_REGEX.match("oci://bucket-name/path/to/file")
+    assert REMOTE_OBJECT_STORE_FILE_REGEX.match("gs://bucket-name/path/to/file")
+    assert REMOTE_OBJECT_STORE_FILE_REGEX.match("dbfs:/Volumes/path/to/file")
+    assert not REMOTE_OBJECT_STORE_FILE_REGEX.match("https://bucket-name/path/to/file")
+    assert not REMOTE_OBJECT_STORE_FILE_REGEX.match("/local/path/to/file")
+    assert not REMOTE_OBJECT_STORE_FILE_REGEX.match("s3:/bucket-name/path/to/file")
+    assert not REMOTE_OBJECT_STORE_FILE_REGEX.match("s3://bucket-name/path/to/file with spaces")
+    assert not REMOTE_OBJECT_STORE_FILE_REGEX.match("s3://bucket-name/path/to/file?")
+
+def test_hf_regex():
+    """Test the regex pattern for Hugging Face dataset paths"""
+    assert HF_REGEX.match("dataset-name")
+    assert HF_REGEX.match("dataset_name")
+    assert HF_REGEX.match("dataset-name_with-mixed.characters-123")
+    assert not HF_REGEX.match("dataset/name")
```

Comment:
er, shouldn't this pass? e.g. mosaicml/dolly_hhrlhf is our default testing dataset